lib/datasource/rubygems/get-gem.ts

```diff
@@ -94,29 +112,31 @@ function isDataStale(): boolean {
   return minutesElapsed >= 5;
 }
 
-let _updateRubyGemsVersions: Promise<void> | undefined;
+let _updateGemVersions: Promise<void> | undefined;
```

Comment:
Shouldn't we use a dictionary indexed by registryUrl here? otherwise it will  can lookup in wrong registry

Reply:
Yes! I already stumbled upon this, see here https://github.com/renovatebot/renovate/pull/6869#issuecomment-665681573 just didn't know how to implement it properly

Reply:
```suggestion
let _updateGemVersions: Record<string, Promise<void> | undefined> = Object.create(null);
```

I think we can simply make a simple dictionary here

Reply:
ok, thats not enough, we need to make `packageReleases` a second level dictionary too.

@rarkins I think we should use the memcache instead a local cache for this?

Reply:
You mean like declaring `packageReleases` as

```
let packageReleases: Record<string, Record<string, string[]>> = Object.create(null);
```

then always access `packageReleases[registryUrl][dependency]`? Can you hint me how to implement it using memcache instead? The current caching mechanism is kinda hard to understand tbh

Reply:
This caching topic is complicated, actually.

Rubygems.org does not fit into the memCache model because it should persist beyond each repository and not re-download it each time.

In the hosted app, any private gemfury instance should not persist past each repository as we don't want to "leak" results. However I think the chanced of it happening is low and the impact is also low (someone could know that there are new versions if they guess the package name correctly.. but that's it I think).

But in a self-hosted setup, the registry should ideally be persistent.